File: apps/docs/examples/snippets/trigger-job.ts
const baseUrl = process.env.STRAIT_BASE_URL ?? "https://api.strait.dev";
const apiKey = process.env.STRAIT_API_KEY;

if (!apiKey) {
  throw new Error("STRAIT_API_KEY is required");
}

const response = await fetch(`${baseUrl}/v1/jobs/send-welcome-email/trigger`, {
  method: "POST",
  headers: {
    Authorization: `Bearer ${apiKey}`,
    "Content-Type": "application/json",
  },
  body: JSON.stringify({
    payload: {
      user_id: "user_abc123",
      email: "customer@example.com",
    },
  }),
});

if (!response.ok) {
  throw new Error(await response.text());
}

const run = await response.json();
console.log(run.id, run.status);

File: apps/docs/examples/snippets/trigger_job.py
import json
import os
import urllib.request


base_url = os.environ.get("STRAIT_BASE_URL", "https://api.strait.dev")
api_key = os.environ["STRAIT_API_KEY"]

body = json.dumps(
    {
        "payload": {
            "user_id": "user_abc123",
            "email": "customer@example.com",
        }
    }
).encode()

request = urllib.request.Request(
    f"{base_url}/v1/jobs/send-welcome-email/trigger",
    data=body,
    method="POST",
    headers={
        "Authorization": f"Bearer {api_key}",
        "Content-Type": "application/json",
    },
)

with urllib.request.urlopen(request, timeout=30) as response:
    run = json.loads(response.read().decode())

print(run["id"], run["status"])
